Todo App v.0.1

NedobezhkinOleksandr opened the pull request:
https://NedobezhkinOleksandr.github.io/react_todo-app/

SerhiiKirik reviewed (changes requested):
Great 🔥 
Let's make some improvements 🧑‍💻 

https://github.com/mate-academy/react_todo-app/assets/91411692/dc3dc0ed-0cef-447d-bee4-8b52ab235b75

NedobezhkinOleksandr commented:
As per review from last mentor:
- Changed colour of text, during renaming todo;
- Border optimized to full-width;
- Loading spinner now individual for all todos, when toggleAll and ClearCompleted;
- Global Spinner added only for first uploading todos from Server;
- Function for deleting todo, when input(during renaming) is empty or with only spaces, left as it is, because the same function in original App(TodoApp);
- Fixed restriction of 'space' on beginning of input(during adding newTodo);

demetrxx reviewed (approved):
Well done 👍 Go ahead!

ivbutsykin commented on src/App.tsx:
```diff
@@ -1,93 +1,243 @@
+
+  const { setIsLoadingAll } = useContext(LoadingContext);
+
+  const isLoading = useRef(false);
```
Wow, why `isLoading` it's ref? Also, I didn't find any places where it's needed, can we remove it?

ivbutsykin commented on src/App.tsx:
```diff
@@ -1,93 +1,243 @@
+
+    const postedTodo = await postTodo(USER_ID, newTodo);
+
+    setTodos((prev: Todo[]) => {
```
```suggestion
    setTodos((prev) => {
```

ivbutsykin commented on src/types/Todo.ts:
```diff
@@ -0,0 +1,11 @@
+  completed: boolean;
+}
+
+export interface PatchedTodo {
```
Can we use sth like this? https://www.typescriptlang.org/docs/handbook/utility-types.html#picktype-keys

ivbutsykin commented on src/utils/fetchClient.ts:
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+const BASE_URL = 'https://mate.academy/students-api';
+
+// returns a promise resolved after a given delay
```
Remove comments such that

ivbutsykin commented on src/index.tsx:
```diff
@@ -1,12 +1,17 @@
-import './styles/index.css';
-import './styles/todo-list.css';
-import './styles/filters.css';
+import 'bulma/css/bulma.css';
```
Do you use it? And probably better to import minified file `import 'bulma/css/bulma.min.css';`
https://couds.github.io/react-bulma-components/?path=/story/welcome--page

ivbutsykin commented on src/components/TodoList/TodoList.tsx:
```diff
@@ -0,0 +1,43 @@
+}) => {
+  return (
+    <div>
+      {todos.map((todo: Todo) => (
```
```suggestion
      {todos.map((todo) => (
```

ivbutsykin commented on src/components/TodoItem/TodoItem.tsx:
```diff
@@ -0,0 +1,166 @@
+    try {
+      setIsLoading(true);
+      await updateTodoTitle(todo.id, newTitle);
+      title = newTitle;
```
It's bad approach. Please find another way to update title in ToDo (if you won't have any ideas, ask in fe_chat)

ivbutsykin reviewed (commented):
Need to fix issues

ivbutsykin reviewed (commented):
lgtm, approved
